pmlc/dialect/tile/transforms/reorder_inputs.cc

```diff
@@ -0,0 +1,128 @@
+// Copyright 2020, Intel Corporation
+
+#include "mlir/IR/Builders.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Support/DebugStringHelper.h"
+#include "llvm/ADT/SetVector.h"
+
+#include "pmlc/dialect/layer/ir/ops.h"
+#include "pmlc/dialect/tile/ir/ops.h"
+#include "pmlc/dialect/tile/transforms/pass_detail.h"
+#include "pmlc/util/layout.h"
+#include "pmlc/util/logging.h"
+
+using namespace mlir; // NOLINT
+
+namespace pmlc::dialect::tile {
+
+namespace {
+
+int64_t getArgPos(Operation &op, BlockArgument &arg) {
+  auto argPos = 0;
+  for (size_t i = 0; i < op.getOperands().size(); i++) {
+    if (op.getOperand(i) == arg) {
+      argPos = i;
+    }
+  }
+  return argPos;
+}
+
+void performDataReordering(OpBuilder &builder, Operation *op, BlockArgument arg,
+                           MLFramework framework, bool isConst) {
+  // Check if op is a BoxOp, if so scan which argument is used,
+  // and get its tensor type.
+  // Later on create the new contraciton op with the same sizes and additional
+  // layout tag. These ops would be moved to init as weights reorders in the
+  // later pass.
+  auto layerOp = dyn_cast<layer::BoxOp>(op);
+  if (!layerOp)
+    return;
+
+  auto argPos = getArgPos(*op, arg);
```

Comment:
Use `arg.getArgNumber()`

Reply:
actually I need a position of the argument in the certain op, not in the function. I did not see a function for that